PVTPackage/source/MultiphaseSystem/PhaseModel/BlackOil/DeadOil_PhaseModel.cpp

```diff
@@ -31,19 +31,25 @@ DeadOil_PhaseModel::DeadOil_PhaseModel( pvt::PHASE_TYPE type,
     m_surfaceMoleDensity( 0 ),
     m_surfaceMolecularWeight( oilSurfaceMw )
 {
-
-  //--Fill table
-  createTable( PVD );
-
-  //Check Consistency
-  checkTableConsistency();
-
-  //Check Consistency
-  checkTableConsistency();
-
-  //Density
-  m_surfaceMassDensity = oilSurfaceMassDensity;
-  m_surfaceMoleDensity = m_surfaceMassDensity / m_surfaceMolecularWeight;
+  // if the phase is present, PVTD.size() > 0
+  // if the phase is absent (for two-phase dead-oil), PVD.size() == 0 => we skip initialization
+  if( PVD.size() > 0 )
```

Comment:
Equivalent question here.

Reply:
Same here. Now it is only used when the phase is present in the system, so I just check the size of PVD[0] and I error out if it is not 3.